typings/styled-components.d.ts

```diff
@@ -51,7 +51,7 @@ export interface ThemedBaseStyledInterface<T> extends ThemedStyledComponentFacto
   <P, TTag extends keyof JSX.IntrinsicElements>(tag: TTag): ThemedStyledFunction<P, T, P & JSX.IntrinsicElements[TTag]>;
   <P, O>(component: StyledComponentClass<P, T, O>): ThemedStyledFunction<P, T, O>;
   <P extends { theme: T; }>(component: React.ComponentClass<P>): ThemedStyledFunction<P, T, WithOptionalTheme<P, T>>;
-  <P>(component: React.ComponentClass<P>): ThemedStyledFunction<P, T>;
+  <P>(component: React.ComponentClass<P> | React.ComponentClass<T>): ThemedStyledFunction<P, T>;
```

Comment:
hmm... it doesn't seem to be correct. This definition says that we can pass a component that takes props in shape of theme data. It doesn't make sense to me.

Reply:
Yeah I agree. I initially went with `any` but following along the interfaces it looked like T props are what the compiler was complaining about.  It looks like React.ComponentClass<P> should be enough for this to match the shape but it doesn't. 
It seems there are some props on T that the compiler is expecting which this union resolves...  

I'm not very familiar with the internals of styled-components but are the styles added as props causing this misshape?

Reply:
let me look closer to the issue you're trying to solve, I may come up with another pr.